mfhdf/test/texternal.c

```diff
@@ -676,10 +676,176 @@ test_special_combos()
     return num_errs;
 } /* test_special_combos() */
 
+/********************************************************************
+   Name: test_change_extdir() - tests re-setting external directory
+                with valid and invalid values.
+   Description:
+    The main contents include:
+    - Create an HDF file and a dataset in it
+    - Set create external directory to the source test directory
+    - Promote the dataset to an external element
+    - Try setting the external directory to various values and verify
+      the behavior of SDreaddata each time
+    - Remove the data file and the external file
+
+   Return value:
+    The number of errors occurred in this routine.
+
+*********************************************************************/
+#define MAIN_FILE "tmainfile.hdf" /* file where the SDS is created */
+#define EXT_FILE  "textfile.txt"  /* file where the external data is stored */
+#define SDS_NAME  "ExternalDS"    /* dataset will have data in an external file */
+#define RANK      1
+#define TMP_DIR   "EXT_tempdir/" /* temporary dir to create the external file in */
+
+static int
+test_change_extdir(void)
+{
+    int32 sd_id;
+    int32 sds_id;
+    float sds1_data[] = {0.1, 2.3, 4.5, 6.7, 8.9};
+    float sds1_out[5];
+    int32 start = 0, stride = 1, edge;
+    int32 dimsize[RANK];
+    char  dir_name[MAX_PATH_LEN]; /* directory from srcdir */
+    intn  dir_name_len;
+    char *another_path      = NULL; /* another path to test reading external file */
+    char *temp_dir          = NULL; /* temp dir to create the external file in */
+    char *created_file_path = NULL; /* path to the created external file */
+    int32 sds_index;
+    int   command_ret = 0; /* retvalue from system commands */
+    intn  status      = 0;
+    intn  num_errs    = 0; /* number of errors in compression test so far */
+
+    status = make_sourcepath(dir_name, MAX_PATH_LEN);
+    CHECK(status, FAIL, "make_datafilename");
+
+    /* When srcdir is not available, make up a directory to create the external
+       file to complicate the subsequent reads */
+    if (!strcmp(dir_name, "./")) {
+
+        /* Facilitate the removal of the temporary directory later */
+        temp_dir = (char *)malloc(strlen(TMP_DIR) + 1);
+        strcpy(temp_dir, TMP_DIR);
+
+#if defined H4_HAVE_WIN32_API && defined __MINGW32__
```

Comment:
Why only mingw with WIN32 API? Why not just WIN32 API?

Reply:
https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/mkdir?view=msvc-170

Reply:
Another search comment: I had to add the include in the if/endif for MinGW lib: #include <direct.h> /* _mkdir */